test/java/seedu/addressbook/storage/StorageFileTest.java

```diff
@@ -0,0 +1,109 @@
+package seedu.addressbook.storage;
+
+import static org.junit.Assert.assertEquals;
+import java.nio.file.Paths;
+import java.util.Collections;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.tag.Tag;
+import seedu.addressbook.data.tag.UniqueTagList;
+import seedu.addressbook.storage.StorageFile.StorageOperationException;
+import static seedu.addressbook.util.TestUtil.assertTextFilesEqual;
+
+public class StorageFileTest {
+    private static final String TEST_DATA_FOLDER = "test/data/StorageFileTest";
+    private static final String TEST_TEMP = "temp.txt";
```

Comment:
I actually don't think this should be a constant as it's not very meaningful. It is just a temporary file name, it can be anything and readers of the code should not need to care (as opposed to `TEST_DATA_FOLDER` where it is important that readers of the code know its location since that is where the test data is stored).